ext/date/php_date.c

```diff
@@ -1033,18 +1039,26 @@ PHP_FUNCTION(strtotime)
 		DATE_TIMEZONEDB, php_date_parse_tzfile_wrapper);
 	error1 = error->error_count;
 	timelib_error_container_dtor(error);
+	if (error1) {
+		timelib_time_dtor(now);
+		timelib_time_dtor(t);
+		RETURN_FALSE;
+	}
+
 	timelib_fill_holes(t, now, TIMELIB_NO_CLONE);
 	timelib_update_ts(t, tzi);
 	ts = timelib_date_to_int(t, &error2);
 
 	timelib_time_dtor(now);
 	timelib_time_dtor(t);
 
-	if (error1 || error2) {
+	/* Seconds since epoch must fit in a zend_long */
+	if (error2) {
```

Comment:
I would possible add a warning, but also rename `error2` to `doesnt_fit` or something better. `error2` says nothing.